internal/stackdriver/stackdriver.go

```diff
@@ -228,31 +237,22 @@ func alignerAndReducer(alignReduceType metrics.AlignReduce) (aligner string, red
 	return
 }
 
-// NewQuery initializes a query.
-func NewQuery(project, region, serviceName, revisionName string) Query {
-	return Query{}.addFilter("resource.labels.project_id", project).
+// newQuery initializes a query.
+func newQuery(project, region, serviceName string) query {
+	return query{}.addFilter("resource.labels.project_id", project).
 		addFilter("resource.labels.location", region).
-		addFilter("resource.labels.service_name", serviceName).
-		addFilter("resource.labels.revision_name", revisionName)
-}
-
-// Query returns the string representation of the query.
-func (q Query) Query() string {
-	return q.filter
+		addFilter("resource.labels.service_name", serviceName)
 }
 
 // addFilter adds a filter to the query.
-func (q Query) addFilter(key, value string) Query {
+//
+// TODO: Support field-based filters, so the query string is generated based on
+// the fields instead of appending a filter everytime this method is called.
+func (q query) addFilter(key, value string) query {
 	if q.filter != "" {
 		q.filter += " AND "
 	}
```

Comment:
doesn't the query end with `" AND"` in this case?

Reply:
No, this case is only when you're adding the first filter. You can see that right after this conditional, the filter is appended

```go
q.filter += fmt.Sprintf("%s=%q", key, value)
```

Let me add a comment in the if statement to make it more clear the case it covers.

Reply:
my bad don't worry